__init__.py

```diff
@@ -340,17 +366,30 @@ def initialize(self):
     def on_websettings_changed(self):
         # Only attempt to load credentials if the username has been set
         # will limit the accesses to the api.
-        if not self.spotify and self.settings.get('user', None):
-            try:
-                self.load_credentials()
-            except Exception:
-                pass
-        if self.spotify:
+        self.log.info('websettings_changed')
+        username = self.settings.get('user', None)
+        password = self.settings.get('password', None)
+        # Username or password has changed
+        if not username:
+            self.handle_error('No username specified')
+            return
+        if not password:
+            self.handle_error('No password specified')
+            return
+        try:
             self.cancel_scheduled_event('SpotifyLogin')
-            if 'user' in self.settings and 'password' in self.settings:
-                if self.process:
-                    self.stop_librespot()
-                self.launch_librespot()
+            self.load_credentials()
+            self.launch_librespot()
+        except Exception as e:
+            self.speak_dialog('Error', data={'error':e})
```

Comment:
This entire method is called in the background when settings change and it's not really a great user experience to have the devices speak error messages out of nowhere.